app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,26 @@
+class SessionsController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.find_by username: params[:user][:username]
+
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "Welcome back #{@user.username}"
+    else
+      @user = User.create username: params[:user][:username]
```

Comment:
You need to check if the `create` method worked!  I tried to log in with a blank username and it said, "Weclome"!